src/c_api/init.cpp

```diff
@@ -27,8 +27,13 @@ void dlaf_initialize(int argc_pika, const char** argv_pika, int argc_dlaf, const
     pika::init_params params;
     params.rp_callback = dlaf::initResourcePartitionerHandler;
     params.desc_cmdline = desc;
+    // After pika 0.21.0 pika::start reports errors only by exception and returns void
+#if PIKA_VERSION_FULL >= 0x001500
```

Comment:
For other reviewers:
```c++
#define PIKA_VERSION_FULL                                                                          \
 ((PIKA_VERSION_MAJOR << 16) | (PIKA_VERSION_MINOR << 8) | PIKA_VERSION_PATCH)
```

Reply:
Yep... not the most intuitive, but we will unfortunately not be changing this in pika simply because of compatibility.